### PR TITLE
NE-150: operator/ingress: Add observedGeneration

### PR DESCRIPTION
**Pull request by Miciah**
https://jira.coreos.com/projects/NE/issues/NE-150

* `operator/v1/types_ingress.go` (`IngressControllerStatus`): Add `ObservedGeneration` field.
* `operator/v1/zz_generated.swagger_doc_generated.go:` Regenerate.

----

@ironcladlou

### CODE REVIEW

**Comment by ironcladlou** [comment]
Hm, shouldn't CRDs have also been regenerated? (https://github.com/openshift/api/pull/470)

**Comment by Miciah** [comment]
> Hm, shouldn't CRDs have also been regenerated? (#470)

Sorry, I'll update.

**Comment by Miciah** [comment]
Maybe I'm misunderstanding, but it looks like operator/v1 CRDs don't get generated, only operator/v1alpha1 and other API groups:  https://github.com/openshift/api/blob/master/Makefile#L24-L29

**Comment by ironcladlou** [comment]
Ah, we will have to fix that up before we can switch over to consuming the CRDs from openshift/api (which I think we should do but which I also think will be a bit of work).

/lgtm

**Comment by ironcladlou** [comment]
@knobunc PTAL for approval

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -351,6 +351,10 @@ type IngressControllerStatus struct {
 	// tlsProfile is the TLS connection configuration that is in effect.
 	// +optional
 	TLSProfile *configv1.TLSProfileSpec `json:"tlsProfile,omitempty"`
+
+	// observedGeneration is the most recent generation observed.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```
Comment (soltysh) [inline operator/v1/types_ingress.go]: The change itself looks good, but you're getting closer to the generic `OperatorStatus` have you considered switching to that?

**Pull request merged**